### PR TITLE
ci: run tests in opam pipeline

### PR DESCRIPTION
**Pull request by jchavarri**
- Consolidates installation of test/dev dependencies between Makefile and ci scripts
- Makes sure tests run in ci on the opam pipeline

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,28 +1,43 @@
-SHELL := $(shell echo $$SHELL)
```
Comment (jchavarri) [inline Makefile:1]: This took me ~1h to figure out, apparently it breaks linux `make` in github actions.

Comment (anmonteiro) [reply Makefile:1]: BUT WHY

File: melange.opam [hunk 2]
```diff
@@ -13,6 +13,7 @@ depends: [
   "cmdliner" {>= "1.1.0"}
   "base64" {>= "3.1.0"}
   "cppo" {build}
+  "reason" {with-test}
```
Comment (anmonteiro) [inline melange.opam:16]: this needs to be written in the `dune-project` file

File: Makefile (hunk 1 above)
Comment (jchavarri) [reply Makefile:1]: idk, you can see the error i got here: https://github.com/melange-re/melange/actions/runs/4365427888/jobs/7634028656#step:6:16

Comment (jchavarri) [reply Makefile:1]: also interestingly, macos works fine 🤷

File: melange.opam (hunk 2 above)
Comment (jchavarri) [reply melange.opam]: done in https://github.com/melange-re/melange/pull/503/commits/d0d7a3f834b850f42890034e2874fdd1c0596229.

Comment (jchavarri) [reply melange.opam]: (i don't like how opam files are generated by dune but they are also tracked by git for publishing purposes, so if I change opam i can see git showing changes and I assume opam is the source, it always confuses me)

File: Makefile (hunk 1 above)
Comment (anmonteiro) [reply Makefile:1]: I thought this was a nice hack. very sad

**Comment by anmonteiro** [comment]
Thanks!

File: Makefile (hunk 1 above)
Comment (jchavarri) [reply Makefile:1]: Just curious, why were you using it for?

**Pull request merged**